cmd/src/main_test.go

```diff
@@ -0,0 +1,337 @@
+			configJsonEnvContents: `{"endpoint": "urn:not-this-one"}`,
+		}
+		expected := config{
+			AccessToken: "",
+			Endpoint:    "urn:this-endpoint-wins",
+		}
+		testCase.Run(func() {
+			runAndAssert(expected, t)
+		})
+	})
+
+	// Endpoint Env | Endpoint in Config(Env) [X] | Endpoint in Flag [X]
+	parentT.Run("endpoint in -config, but -endpoint wins", func(t *testing.T) {
+		testCase := FlagTestCaseInput{
+			testState: t,
+
+			endpointFlag:          "urn:look-at-me-instead",
+			configJsonEnvContents: `{"endpoint": "urn:nothing-to-see-here"}`,
+		}
+		expected := config{
+			AccessToken: "",
+			Endpoint:    "urn:look-at-me-instead",
+		}
+		testCase.Run(func() {
+			runAndAssert(expected, t)
+		})
+	})
+
+	parentT.Run("ensure env-vars do not leak", func(t *testing.T) {
+		assertEquals(os.Getenv(AccessTokenEnvVar), "", t)
+		assertEquals(os.Getenv(ConfigEnvVar), "", t)
+		assertEquals(os.Getenv(EndpointEnvVar), "", t)
+	})
+}
+
+// will run my setup logic, then the provided function, and then clean up after myself
+// any encountered `error`s are handled via the testing.T functions for that purpose
+func (testCase *FlagTestCaseInput) Run(inner func()) {
+	cleanupFuncs := testCase.setUp()
+	inner()
+	for _, fn := range cleanupFuncs {
+		if e := fn(); e != nil {
+			testCase.testState.Fatal(e)
+		}
+	}
+}
+
+// Evaluate this test case input, which may involve altering the `os.Setenv`
+// but the returned callbacks are cleanup functions that will tear down any changes made by setUp
+func (testCase *FlagTestCaseInput) setUp() []func() error {
+	// any post-test cleanup action
+	var cleanupFuncs []func() error
+
+	if testCase.accessTokenEnv != "" {
+		cleanup := pushSetenv(AccessTokenEnvVar, testCase.accessTokenEnv)
+		cleanupFuncs = append(cleanupFuncs, cleanup)
+	}
+	if testCase.endpointEnv != "" {
+		cleanup := pushSetenv(EndpointEnvVar, testCase.endpointEnv)
+		cleanupFuncs = append(cleanupFuncs, cleanup)
+	}
+
+	if testCase.endpointFlag != "" {
+		oldEndpoint := endpoint
+		endpoint = &testCase.endpointFlag
+		cleanupFuncs = append(cleanupFuncs, func() error {
+			endpoint = oldEndpoint
+			return nil
+		})
+	}
+
+	if testCase.configJsonEnvContents != "" {
+		configFile := writeTempConfigFile(testCase.configJsonEnvContents, testCase.testState)
+		configFilename := configFile.Name()
+		cleanup := pushSetenv(ConfigEnvVar, configFilename)
+		cleanupFuncs = append(cleanupFuncs, cleanup)
+		cleanupFuncs = append(cleanupFuncs, func() error {
+			if e := os.Remove(configFilename); e != nil {
+				return e
+			}
+			return nil
+		})
+	}
+
+	if testCase.configJsonFlagContents != "" {
+		configFile := writeTempConfigFile(testCase.configJsonFlagContents, testCase.testState)
+		configFilename := configFile.Name()
+		oldConfigPath := configPath
+		configPath = &configFilename
+		cleanupFuncs = append(cleanupFuncs, func() error {
+			// cheat and piggy back on this cleanup
+			configPath = oldConfigPath
+
+			if e := os.Remove(configFilename); e != nil {
```

Comment:
```go
return os.Remove(configFilename)
```

Reply:
Damn, you caught me napping on that one :-( it was fallout from some refactoring that moved the code but not the neurons. I'll straighten it out